update/repository.go

```diff
@@ -0,0 +1,159 @@
+package update
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"fmt"
+	"github.com/pkg/errors"
+	"strings"
+	"time"
+)
+
+const TenantsUpdateTableName = "tenants_update"
+
+type Status string
+
+const (
+	Finished Status = "finished"
+	Updating Status = "updating"
+	Failed   Status = "failed"
+)
+
+// Value - Implementation of valuer for database/sql
+func (s Status) Value() (driver.Value, error) {
+	return string(s), nil
+}
+
+// Scan - Implement the database/sql scanner interface
+func (s *Status) Scan(value interface{}) error {
+	if value == nil {
+		*s = Status("")
+		return nil
+	}
+	if bv, err := driver.String.ConvertValue(value); err == nil {
+		// if this is a bool type
+		if v, ok := bv.(string); ok {
+			// set the value of the pointer yne to YesNoEnum(v)
+			*s = Status(v)
+			return nil
+		}
+	}
+	// otherwise, return an error
+	return errors.New("failed to scan status")
+}
+
+type Repository interface {
+	GetStatus() (Status, error)
+	UpdateStatus(status Status) error
+	GetLastTimeUpdated() (time.Time, error)
+	UpdateLastTimeUpdated() error
+	PrepareForUpdating() error
+	GetFailedCount() (int, error)
+	IncrementFailedCount() error
+	IsVersionSame(fileVersionAttrName string, expectedVersion string) (bool, error)
+	UpdateVersionsTo(attrNameMapping map[string]*VersionWithTypes) error
+}
+
+type GormRepository struct {
+	tx *sql.Tx
+}
+
+func NewRepository(tx *sql.Tx) *GormRepository {
+	return &GormRepository{
+		tx: tx,
+	}
+}
+
+func (r *GormRepository) GetStatus() (Status, error) {
+	query := fmt.Sprintf("SELECT status FROM %s", TenantsUpdateTableName)
```

Comment:
Just out of curiosity why you are using a lot of SQL raw stuff instead of GORM. Any particular reason?

Reply:
added test here 0d2b231  I covered both cases there - when tenants are successfully updated and when the update failed

Reply:
as part of the commit, there is a change of the command that starts the execution of integration tests: https://github.com/fabric8-services/fabric8-tenant/commit/b0721b02713d14073c6db94c585e20aa854c874c#diff-552b9cf9c536f6f705ab73d1ab26b005R184
The reason is that some tests were failing only on my machine from the very first moment (on jenkins and on @xcoulon PC the build was green) and the tests were obviously influenced by each other. I took me a very long time to figure out that the problem was parallel test execution (even though that `t.Parallel()` is not used), that was caused by parallel builds - for more information see https://coderwall.com/p/rwgo_a/go-test-package-in-parallel

Reply:
@xcoulon, @MatousJobanek wanna add this `-p 1` https://github.com/fabric8-services/fabric8-tenant/commit/b0721b02713d14073c6db94c585e20aa854c874c#diff-552b9cf9c536f6f705ab73d1ab26b005R184 flag to all our repos (+fabric8-starter if relevant)? :)